### PR TITLE
updated sonarqube deployment template that includes a jenkins web hook

### PR DESCRIPTION
**Pull request by haithamshahin333**
Included a post hook in order to build a webhook into sonarqube. @dwasinge @sherl0cks

### CODE REVIEW

**Review by InfoSec812 — COMMENTED** [review]
This looks like a GREAT addition, but why a separate file? Why not just put this in the existing template?

**Comment by haithamshahin333** [comment]
@InfoSec812 thanks for the review. I can go ahead and make it part of the standard sonarqube template and make a new PR

**Comment by sherl0cks** [comment]
No need to make a new one @haithamshahin333 , just update this branch. Also, so comments inline.

File: sonarqube/sonarqube-deployment-post-hook-template.yml [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+    strategy:
+      activeDeadlineSeconds: 21600
+      recreateParams:
+        timeoutSeconds: 600
+        post:
+          execNewPod:
+            command:
+            - /bin/sh
+            - -c
+            - sleep 30 && curl http://admin:admin@sonarqube:9000/api/webhooks/create
```
Comment (sherl0cks) [inline sonarqube/sonarqube-deployment-post-hook-template.yml]: I'm curious why the sleep is needed if there is already a readiness check. Does post not wait for a ready state?

Comment (haithamshahin333) [reply sonarqube/sonarqube-deployment-post-hook-template.yml]: @sherl0cks in my testing it didn't seem that the readiness check was applied to the post hook container. Also check out this blog - https://blog.openshift.com/using-post-hook-to-initialize-a-database/. They similarly add some delay for the container to come up before executing the actual hook.

Comment (sherl0cks) [reply sonarqube/sonarqube-deployment-post-hook-template.yml]: OK

**Comment by sherl0cks** [comment]
@haithamshahin333 looks good. can you resolve the merge issues? If not, a new PR is fine

**Comment by haithamshahin333** [comment]
Hey @pcarney8 - curious as to why the sonarqube deployment templates were removed. May help me to understand why before merging this into the repo.

**Comment by sherl0cks** [comment]
Closing this. The conflict is because sonarqube stuff was moved to https://github.com/redhat-cop/containers-quickstarts/blob/master/sonarqube/.openshift/templates/sonarqube-deployment-template.yml, and thus these changes belong there.

**Pull request closed without merging**